src/escalus_bosh.erl

```diff
@@ -363,8 +363,14 @@ handle_call(mark_as_terminated, _From, #state{} = State) ->
 
 handle_call(get_active, _From, #state{active = Active} = State) ->
     {reply, Active, State};
-handle_call({set_active, Active}, _From, State) ->
-    {reply, ok, State#state{active = Active}};
+handle_call({set_active, Active}, _From, #state{replies = Replies} = State) ->
+    NewState = if
```

Comment:
We usually try to avoid `if` statements in Erlang. Please rewrite it a `case` statement, ideally in a separate function returning the state.